app/assets/javascripts/Components/submission_file_manager.jsx

```diff
@@ -254,9 +258,67 @@ class SubmissionFileManager extends React.Component {
     );
   };
 
+  renderRequiredFiles = () => {
+    let requiredFilesBox;
+
+    if (this.state.requiredFiles.length > 0) {
+      requiredFilesBox = (
+        <div>
+          <h2>{I18n.t("activerecord.attributes.assignment.assignment_files")}</h2>
+          <p>
+            {this.state.numberOfMissingFiles === 0 ? (
+              <p>{I18n.t("student.submission.all_files_submitted")}</p>
+            ) : (
+              I18n.t("student.submission.missing_files", {file: this.state.numberOfMissingFiles})
+            )}
+          </p>
+          {this.state.requiredFiles.map(filename => {
+            return (
+              <div key={filename} className={"required-files-container"}>
```

Comment:
What I meant earlier is that this `div` should instead be a `p` (and see my comment on the inner `p` you added below). That will make the spacing a bit nicer.